### ISSUE TITLE
video-sync=display-resample with vo_dmabuf_wayland causes a crash after a few minutes

### ISSUE THREAD

**Issue by llightcb** [mpv-player/mpv#11773]
### Important Information

mpv --version:
>mpv 0.35.0-UNKNOWN Copyright © 2000-2023 mpv/MPlayer/mplayer2 projects
 built on UNKNOWN
libplacebo version: v5.264.1
FFmpeg version: 6.0
FFmpeg library versions:
   libavutil       58.2.100
   libavcodec      60.3.100
   libavformat     60.3.100
   libswscale      7.1.100
   libavfilter     9.3.100
   libswresample   5.0.100

apk info mpv:
>mpv-0.35.1_git20230605-r0 

see: [new abort](https://git.alpinelinux.org/aports/commit/?id=190c146c766bf5bc7e241d236fb833c342db044c)

> alpine linux v3.18

>6.1.32-0-lts

>sway-1.8.1

> wlroots-0.16.2

> intel uhd graphics 620 (rev 07)
i915 v. kernel
EGL/GBM

### Reproduction steps

mpv --no-config --video-sync=display-resample --vo=dmabuf-wayland [file]

### Expected behavior

no crash.
 
### Actual behavior

mpv chrashes after a few minutes with error:

```
wl_display@1: error 1: invalid arguments for zwp_linux_buffer_params_v1@34.add
[vo/dmabuf-wayland/wayland] Error occurred on the display fd, closing
```

no crash before the switch to 0.35.1_git20230605 .  whether there is a causal connection to this is unclear to me.

### Log file

[log](https://0x0.st/HcKt.txt)

**Comment by Dudemanguy** [comment]
Can you try mpv from git master? This recently got a rewrite and it seems you're using 0.35 from the log.

**Comment by Dudemanguy** [comment]
It's commit c1de4bb7454db8fbb61e337b58923652602524dd that caused this. Now why this only happens with `display-resample` is a good question (somehow adding params when we shouldn't be?).

**Comment by llightcb** [comment]
well, this was fast :+1: .  the commit is interesting. unfortunately, i don't have a brain as big as devs, so i'm at a loss. but having identified the commit that probably leads to this behavior, and your remark, give me hope. 

o.t.: regarding the "version-thing" on irc: it's ok. they say, "the VERSION has that string in it". it will be overwriten.

**Comment by nekopsykose** [comment]
now it will say `mpv 0.35.1-effc68063bc3b4c856d70b8c6800d5bdb072815c`, i guess you're meant to override it since nothing else seems to touch it :)

**Comment by llightcb** [comment]
psykose! i love you. know that.

**Comment by Dudemanguy** [comment]
@llightcb: by the way, did display-resample actually *ever* work for you? I can "fix" this but what will happen is an awful flickering artifact because the way the compositor holds onto the buffer while mpv switches image. I double checked before https://github.com/mpv-player/mpv/commit/c1de4bb7454db8fbb61e337b58923652602524dd and it had the exact same issue.

**Comment by Dudemanguy** [comment]
No I mean did dmabuf-wayland + display-resample have a flickering artifact for you in the past? Depending on the video you play, you should have seen flickering behavior where it rapidly appears to flicker black. This happened before the rewrite commit. Currently in master, this doesn't flicker *but* instead it will eventually crash because I think the source image of the buffer doesn't match the params given. Now with the small patch I linked, it fixes the crash but it makes it behave like it did before in the past: by reintroducing that flickering behavior.

tl;dr: I don't think dmabuf-wayland and display-resample ever actually worked or at least not for me but I'm not 100% sure. If you see different behavior somehow, that would be worth knowing.

**Comment by llightcb** [comment]
and back again. i pinned the `v3.18/community` repo, uninstalled my mpv version from `edge` and installed the previous one (before the switch on edge) from git master. no crash - of course. regarding video quality:  i tested some videos; with or without `display-resample`. i could not detect any _additional_ flickering/artifacts with `vo=dmabuf-wayland` + `display-resample` .  of course, i watched [big_buck_bunny_720p_30mb.mp4](https://sample-videos.com/) too -  closely. same result (as far as my eyes don't cloud me..). where exactly in the linked video file can you detect the described "flicker black"? at which point? so we could make a straight comparison. as additional information, my mpv [config](https://github.com/llightcb/sight/blob/main/home/.mpv/mpv.conf). only difference @ home: `vo=gpu` is disabled, `vo=dmabuf-wayland` is enabled.

**Comment by llightcb** [comment]
done! i don't have too many videos on my machine, but none of those videos (mkv/mp4/vp9 et cetera) showed your described obvious problems ("you know it when it happens") with dmabuf-wayland and display-resample. i guess i was just lucky.

anyway: nekopsykose built it with mrtest from pipeline artifacts for me, so i only had to execute `mrtest add -a xyz` in order to test your code. everything perfect, runs smoothly, low cpu-usage even with `sharpen` (the main reason why i prefer vo_dmabuf_wayland), good quality, no crash. 

i guess you have solved the problem. thank you for the time and brain-work you have put into it. i love you too - of course ;).

**Comment by Dudemanguy** [comment]
Thanks for reporting it!
